molgenis-security/src/main/java/org/molgenis/security/permission/UserPermissionEvaluatorImpl.java

```diff
@@ -15,16 +17,21 @@
 public class UserPermissionEvaluatorImpl implements UserPermissionEvaluator
 {
 	private final PermissionEvaluator permissionEvaluator;
+	private final MutableAclService mutableAclService;
+	private final UserAccountService userAccountService;
 
-	UserPermissionEvaluatorImpl(PermissionEvaluator permissionEvaluator)
+	UserPermissionEvaluatorImpl(PermissionEvaluator permissionEvaluator, MutableAclService mutableAclService,
+			UserAccountService userAccountService)
 	{
 		this.permissionEvaluator = requireNonNull(permissionEvaluator);
+		this.mutableAclService = requireNonNull(mutableAclService);
+		this.userAccountService = requireNonNull(userAccountService);
 	}
 
 	@Override
 	public boolean hasPermission(ObjectIdentity objectIdentity, Permission permission)
 	{
-		if (SecurityUtils.currentUserIsSuOrSystem())
+		if (SecurityUtils.currentUserIsSuOrSystem() || isOwner(objectIdentity))
```

Comment:
Is this the place to check this? has the owner of the ACL always full permissions on an object?